joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreatorHelper.scala

```diff
@@ -0,0 +1,124 @@
+package io.joern.rubysrc2cpg.astcreation
+import io.joern.rubysrc2cpg.astcreation.GlobalTypes.{builtinFunctions, builtinPrefix}
+import io.joern.rubysrc2cpg.parser.ParserAst.ParserNode
+import io.joern.x2cpg.datastructures.Scope
+import io.joern.x2cpg.datastructures.Stack.*
+import io.shiftleft.codepropertygraph.generated.PropertyNames
+import io.shiftleft.codepropertygraph.generated.nodes.*
+
+trait AstCreatorHelper { this: AstCreator =>
+
+  /* Used to track variable names and their LOCAL nodes.
+   * TODO: Perhaps move this feature into a new Pass?
+   */
+  protected val scope: Scope[String, NewNode, NewNode] = new Scope()
+
+  /* Used to compute a method's full name and parent.
+   * TODO: port RubyScope from the deprecated frontend here?
+   * */
+  protected val methodAstParentStack: Stack[NewNode] = new Stack()
+
+  protected def getEnclosingAstType: String     = methodAstParentStack.head.label()
+  protected def getEnclosingAstFullName: String = methodAstParentStack.head.properties(PropertyNames.FULL_NAME).toString
+  protected def computeClassFullName(name: String): String  = s"$getEnclosingAstFullName.$name"
+  protected def computeMethodFullName(name: String): String = s"$getEnclosingAstFullName:$name"
+
+  override def column(node: ParserNode): Option[Integer]    = node.column
+  override def columnEnd(node: ParserNode): Option[Integer] = node.columnEnd
+  override def line(node: ParserNode): Option[Integer]      = node.line
+  override def lineEnd(node: ParserNode): Option[Integer]   = node.lineEnd
+
+  protected def isBuiltin(x: String): Boolean      = builtinFunctions.contains(x)
+  protected def prefixAsBuiltin(x: String): String = s"$builtinPrefix$pathSep$x"
+  protected def pathSep                            = "."
+}
+
+// TODO: Move this to a more appropriate place?
+object GlobalTypes {
+  val builtinPrefix = "__builtin"
+
+  /* Sources:
+   * https://ruby-doc.org/docs/ruby-doc-bundle/Manual/man-1.4/function.html
+   * https://ruby-doc.org/3.2.2/Kernel.html
+   *
+   * We comment-out methods that require an explicit "receiver" (target of member access.)
```

Comment:
> We comment-out methods that require an explicit "receiver" (target of member access.)

Are these inherited by some top-level object? Like Java's `java.lang.Object`? Might be useful to stub these, though just a thought for future.